Assets/Scripts/Enemy/EnemyBehaviour.cs

```diff
@@ -47,7 +48,9 @@ public void TakeDamage(float Damage)
                 score++;
                 PlayerPrefs.SetFloat("Score", score);
                 //Enemy Death state
-                Destroy(gameObject);
+                //return the enemy to the pool
+                this.gameObject.SetActive(false);
+                EnemyPoolService.Instance.ReturnPooledObject(this.gameObject);
```

Comment:
return a specific Controller/view to the pool not the game object

Reply:
the pool should be of the controller or the view - never just game objects

Reply:
Done added view